### ISSUE TITLE
Css autocompilation in wrong directory

### ISSUE THREAD

**Issue by mdczaplicki** [jrief/django-sass-processor#110]
Whenever I runserver and render any template with `sass_src` tag in it, it does actually compile my scss file to css, but in `STATIC_URL` instead of `STATIC_ROOT`.
So on every change of my scss file I have to run `./manage.py compilescss` and then it is compiled next to scss file (which is in `STATIC_ROOT`).
I've read this and it should be done automatically https://github.com/jrief/django-sass-processor/issues/55

**Comment by jrief** [comment]
when `DEBUG is True` this is the expected behavior. In all of my setups, it works like this.
You then certainly have a configuration error. Please compare against the relevant parts [here](https://github.com/awesto/django-shop/blob/master/example/myshop/settings.py). This setup is known to work.

**Comment by mdczaplicki** [comment]
This is my settings.py and it seems that there is the same config as in mine
`
"""
Django settings for JaguarWedding project.

Generated by 'django-admin startproject' using Django 2.1.1.

For more information on this file, see
https://docs.djangoproject.com/en/2.1/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/2.1/ref/settings/
"""

import os

# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))


# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/2.1/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = True

ALLOWED_HOSTS = ['localhost', '0.0.0.0', '127.0.0.1', '192.168.0.6']


# Application definition

INSTALLED_APPS = [
    'multi_captcha_admin',
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'main.apps.MainConfig',
    'easy_thumbnails',
    'snowpenguin.django.recaptcha2',
    'sass_processor',
]

MIDDLEWARE = [
    'django.middleware.security.SecurityMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
]

ROOT_URLCONF = 'JaguarWedding.urls'

TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [os.path.join(BASE_DIR, 'templates')]
        ,
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
                'main.context_processors.strings',
            ],
        },
    },
]

WSGI_APPLICATION = 'JaguarWedding.wsgi.application'


# Database
# https://docs.djangoproject.com/en/2.1/ref/settings/#databases

DATABASES = {
    'default': {
    }
}


# Password validation
# https://docs.djangoproject.com/en/2.1/ref/settings/#auth-password-validators

AUTH_PASSWORD_VALIDATORS = [
    {
        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
    },
]


# Internationalization
# https://docs.djangoproject.com/en/2.1/topics/i18n/

LANGUAGE_CODE = 'pl-pl'

TIME_ZONE = 'UTC'

USE_I18N = True

USE_L10N = True

USE_TZ = True


# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/2.1/howto/static-files/

STATIC_ROOT = os.path.join(BASE_DIR, 'static')
STATIC_URL = '/static/'
STATICFILES_DIRS = [
    'templates/lib',
    'templates/static'
]

MEDIA_ROOT = os.path.join(BASE_DIR, 'media')
MEDIA_URL = '/media/'

THUMBNAIL_ALIASES = {
    '': {
        'parallax': {'size': (1200, 510), 'crop': True},
        'gallery': {'size': (266, 150), 'crop': True}
    }
}

MULTI_CAPTCHA_ADMIN = {
    'engine': 'recaptcha2',
}

RECAPTCHA_PRIVATE_KEY = '6LdEcnIUAAAAAPeCFoUM4MNWQdpHBHPDfVpxN5HP'
RECAPTCHA_PUBLIC_KEY = '6LdEcnIUAAAAAHRjJ1eulgtUKbbIfI547ADMyl_8'

FILE_UPLOAD_PERMISSIONS = 0o777;

STATICFILES_FINDERS = [
    'django.contrib.staticfiles.finders.FileSystemFinder',
    'django.contrib.staticfiles.finders.AppDirectoriesFinder',
    'sass_processor.finders.CssFinder',
]`

**Comment by mdczaplicki** [comment]
I'm not sure I follow. There is a file `main.css` next to `main.scss`. And this file is only updated when I run `compilescss`. So when I am during development in my IDE I have to run `compilescss` everytime I make a change in .scss file to see the changes in my browser.   
My .scss file is in STATIC_ROOT directory.

**Comment by jrief** [comment]
> So when I am during development in my IDE I have to run compilescss everytime I make a change in .scss file to see the changes in my browser.

No, absolutely not. This would thwart the whole idea of django-sass-processor.

You should put your .scss files into a directory below your project in, say `static/myproject/css/style.scss`. This directory is **not** your STATIC_ROOT. It is managed by your VCS. STATIC_ROOT point somewhere else and is not managed by your VCS. Into there django-sass-processor puts the compiled files. When the browser then load the compiled .css-file, Django looks for it using the `FileSystemFinder`.

**Comment by mdczaplicki** [comment]
![image](https://user-images.githubusercontent.com/9108586/46599985-0e3efd80-cae9-11e8-8789-f3f9e0c68c81.png)
Ok, my bad, I indeed hadn't my *.scss file in STATIC_ROOT. So what do I have wrong in my setup? Could you tell by looking on this screenshot only?

**Comment by jrief** [comment]
invoke `./manage.py compilescss --delete-files`. This should remove the lower `branding.css` file. Afterwards try to load http://localhost:8000/static/css/branding.css , this should load the upper `branding.css`. If that doesn't work, your Django setup is screwed. You then have to figure out yourself, because this is pure Django functionality.

**Comment by mdczaplicki** [comment]
It does indeed delete css file next to scss file to which http://localhost:8000/static/css/branding.css directs, but we I visit this url I get 404, and when I render template it creates css file in project root/static/css/branding.css to which nothing points.
I'm having hard time believing I hade something screwd up in project. It happens in two of my projects. 
Maybe somehow your library is not compatible with django 2.1 python 3.6?

**Comment by jrief** [comment]
>  it creates css file in project root/static/css/branding.css to which nothing points.

And from here on, everything is pure Django. As I told you before, you have to find out how to serve this file using Django's `./manage.py runserver` with `DEBUG = True`. Read the Django docs or ask on Stackoverflow.

**Comment by mdczaplicki** [comment]
Ok, thank you very much. There was zero aggression in my previous post as it may seem like it was aggressive. Sorry, have a nice day :)
